connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryUtil.scala

```diff
@@ -47,6 +48,28 @@ object BigQueryUtil {
   val QUALIFIED_TABLE_REGEX: Regex =
     s"""^((($PROJECT_PATTERN)[:.])?($DATASET_PATTERN)\\.)?($TABLE_PATTERN)$$""".r
 
+  private def dtoTableSchemaToBqSchema(dtoSchema: TableSchema): Schema = {
+    val fromPbMethod =
+      classOf[Schema]
+        .getDeclaredMethods
+        .toIterable
+        .find(method => method.getName == "fromPb")
+        .get
+
+    fromPbMethod.setAccessible(true)
+    fromPbMethod.invoke(null, dtoSchema).asInstanceOf[Schema]
+  }
+
+  def parseSchemaFromJson(jsonSchemaString: String): Schema = {
+    val tableFieldSchema = PARSER
```

Comment:
could there be exceptions in parsing here? How will they be handled?

Reply:
I'd do following: make a return type as Option[Schema] so if there're errors None will be returned and error will be logged, otherwise Some(schema)
Let me know if this makes sense or maybe I can handle this some other way